pkg/agent/openflow/feature_templates.go

```diff
@@ -0,0 +1,369 @@
+//                                  |                     \                       /
+//                                  |                      \                     /
+//                     +------------------------+        +-------------------------+
+//                     |        Pipeline        |        |         Pipeline        |
+//                     +------------------------+        +-------------------------+
+//
+// FeatureTable:
+// FeatureTable in Flexible Pipeline is like "iptables entry" in iptables. It is the basic unit to build OVS pipelines.
+//
+// Stage:
+// Stage in Flexible Pipeline is like "chain" in iptables. The order of Stages is fixed. A Stage can be empty or have
+// one or more FeatureTables. Note that, a FeatureTable cannot be shared by different Stages. All Stages include,
+//   - ClassifierStage
+//	 - ValidationStage
+//	 - ConntrackStateStage
+//	 - PreRoutingStage
+//	 - EgressSecurityStage
+//	 - RoutingStage
+//	 - PostRoutingStage
+//	 - SwitchingStage
+//	 - IngressSecurityStage
+//	 - ConntrackStage
+//	 - OutputStage
+//
+// Feature:
+// Feature in Flexible Pipeline is like "table" in iptables. A Feature is intended to implement a major function. Note
+// that, a Feature can have one or more Stages, and a Stage can be shared by multiple Features. All Features include,
+//   - PodConnectivity, implementation of connectivity for Pods
+//   - NetworkPolicy, implementation of K8s NetworkPolicy and Antrea NetworkPolicy
+//   - Service, implementation of K8s Service
+//   - Egress, implementation of Egress
+//   - Traceflow, implementation of Traceflow
+//   - Multicast, implementation of Multicast
+//
+// Pipeline:
+// This is a concept in Flexible Pipeline, not OVS pipeline, but a corresponding OVS pipeline will be realized in OVS
+// finally. A Pipeline is intended to implement a set of major functions. Note that, a Pipeline can have one or more
+// Features, but a Feature can be shared by multiple Pipelines. All Pipelines include,
+//   - Pipeline for IP
+//   - Pipeline for ARP
+//   - Pipeline for multicast
+//
+// How to insert a flow table with framework Flexible Pipeline?
+// - Select a target Pipeline, Feature and Stage.
+// - Define a FeatureTable with table name, Stage and priority. Arrange the FeatureTable in desired position with appropriate
+//   priority value. For example, there is a Stage A which has FeatureTable A1 (priority 100) and FeatureTable A2 (priority 200)
+//   declared by other Features. To insert a FeatureTable A3:
+//     - between A1 and A2, the priority of A3 should have a priority between 101 to 199
+//     - after A1, the priority of A3 should be less than 100
+//     - before A2, the priority of A3 should be greater than 200
+// - Reference the defined FeatureTable in target Feature. See method `getTemplate(p pipeline) *featureTemplate`.
+
+// FeatureTable is just the concept of FeatureTable in framework Flexible Pipeline. If a FeatureTable is used by one
```

Comment:
FeatureTable is just the concept of FeatureTable in framework Flexible Pipeline. -> struct FeatureTable refers to the term "FeatureTable" described in framework Flexible Pipeline.

Reply:
Thanks.